File: LMS.App/ClientApp/src/root/interfaces/notification/notificationViewModel.ts
export interface NotificationViewModel
{
    id:string,
    userId:string,
    actionDoneBy:string,
    avatar:string,
    isRead:boolean,
    notificationContent:string,
    notificationType:NotificationType,
    postId?:string | null,
    postType?:number,
    post?:any,
    reelId?:string | null,
    chatType?:number | null,
    chatTypeId?:string| null,
    meetingId?:string | null,
    followersIds?:string[] | null

}

export enum NotificationType{
    Likes = 1,
    Followings = 2,
    StatusChanges = 3,
    LectureStart = 4,
    Messages = 5,
    AssignTeacher = 6,
    CertificateSent = 7,
    Transaction = 8,
    Report = 9
}

File: LMS.Common/ViewModels/Notification/NotificationViewModel.cs
using LMS.Common.ViewModels.Post;
using LMS.Common.ViewModels.User;
using LMS.Data.Entity;
using LMS.Data.Entity.Chat;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LMS.Common.ViewModels.Notification
{
    public class NotificationViewModel
    {
        public Guid Id { get; set; }
        public string? UserId { get; set; }
        public string? ActionDoneBy { get; set; }
        public UserDetailsViewModel? User { get; set; }
        public string? Avatar { get; set; }
        public Boolean IsRead { get; set; }
        public DateTime? DateTime { get; set; }
        public string NotificationContent { get; set; }
        public NotificationTypeEnum NotificationType { get; set; }
        public Guid? PostId { get; set; }
        public PostDetailsViewModel? Post { get; set; }
        public int? PostType { get; set; }
        public string? ReelId { get; set; }
        public ChatType? ChatType { get; set; }
        public Guid? ChatTypeId { get; set; }
        public string? MeetingId { get; set; }
        public List<string>? FollowersIds { get; set; }
    }
}
